### ISSUE TITLE
Mongoid Support

### ISSUE THREAD

**Issue by smarquez1** [lynndylanhurley/devise_token_auth#15]
Including the model concern 'DeviseTokenAuth::Concerns::User' on a Mongoid model returns the folowing error message:

~/.rvm/gems/ruby-2.1.2@tp/gems/devise_token_auth-0.1.25/app/models/devise_token_auth/concerns/user.rb:11:in `block in <module:User>': undefined method`serialize' for User:Class (NoMethodError)

**Comment by iDuuck** [comment]
@austiine04 For me the issue of this error was, that I put the Concern **before** the `include Mongoid::Document` instead of **behind it**. Put it behind and you will have the same problem as the others here.

**Comment by dreamermansoor** [comment]
Are we seeing this feature implemented in the gem anytime soon ?  It works for me using @ybian  workaround but will be great if we have mongodb support

**Comment by ulitiy** [comment]
Happy birthday to this issue. Making the gem unusable.

**Comment by booleanbetrayal** [comment]
You can submit a PR @ulitiy. That's the beauty of open source software.

**Comment by vs4vijay** [comment]
Managed to work with Mongoid, @ybian Thanks. If anyone wanna know, take a look at my repo. https://github.com/vs4vijay/carzilla

**Comment by eugenemelnikov** [comment]
Any progress with Mongoid support?

**Comment by kirykr** [comment]
@ybian, @vs4vijay 
```
Fetching gem metadata from https://rubygems.org/.........
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies....
Bundler could not find compatible versions for gem "rails":
  In snapshot (Gemfile.lock):
    rails (= 5.1.4)

  In Gemfile:
    rails (~> 5.1.4)

    devise_token_auth was resolved to 0.1.30.beta3, which depends on
      rails (~> 4.2)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```
Can we make it compatible with rails 5.1.x

**Comment by MaicolBen** [comment]
closed in favor of #1198
